numpy/linalg/tests/test_linalg.py

```diff
@@ -1583,8 +1583,45 @@ def check_qr(self, a):
         assert_almost_equal(r2, r1)
 
     def test_qr_empty(self):
-        a = np.zeros((0, 2))
-        assert_raises(linalg.LinAlgError, linalg.qr, a)
+        for m, n in [(3, 0), (0, 3)]:
+            k = min(m, n)
+            a = np.empty((m, n))
+            a_type = type(a)
+            a_dtype = a.dtype
+            #
+            q, r = np.linalg.qr(a, mode='reduced')
+            assert_(q.dtype == a_dtype)
+            assert_(r.dtype == a_dtype)
+            assert_(isinstance(q, a_type))
+            assert_(isinstance(r, a_type))
+            assert_(q.shape == (m, k))
+            assert_(r.shape == (k, n))
+            assert_almost_equal(np.dot(q, r), a)
+            assert_almost_equal(np.dot(q.T.conj(), q), np.eye(k))
+            assert_almost_equal(np.triu(r), r)
```

Comment:
I'm not sure these three tests are interesting - in these tests `k == 0`, `a.size == 0`, and `r.size == 0` - so there are no values to compare anyway

Reply:
I'll drop some of the ones that are implied

Reply:
Actually - just call `self.check_qr(a)`, and it will do all the work for you